dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,770 @@
+                this._vectorTableName,
+                this._propertyReader.KeyPropertyStoragePropertyName,
+                this._vectorTableStoragePropertyNames.Value,
+                storageModels);
+
+            await vectorInsertCommand.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        using var dataCommand = this._commandBuilder.BuildInsertCommand(
+                this._dataTableName,
+                this._propertyReader.KeyPropertyStoragePropertyName,
+                this._dataTableStoragePropertyNames.Value,
+                storageModels,
+                replaceIfExists: true);
+
+        using var reader = await dataCommand.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+
+        while (await reader.ReadAsync(cancellationToken).ConfigureAwait(false))
+        {
+            var key = reader.GetFieldValue<TKey?>(0);
+
+            if (key is not null)
+            {
+                yield return key;
+            }
+
+            await reader.NextResultAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    private Task InternalDeleteAsync<TKey>(
+        TKey key,
+        DeleteRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        Verify.NotNull(key);
+
+        var condition = new SqliteWhereEqualsCondition(this._propertyReader.KeyPropertyStoragePropertyName, key);
+
+        return this.InternalDeleteBatchAsync([key], condition, options, cancellationToken);
+    }
+
+    private Task InternalDeleteBatchAsync<TKey>(
+        IEnumerable<TKey> keys,
+        DeleteRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        var condition = new SqliteWhereInCondition(
+            this._propertyReader.KeyPropertyStoragePropertyName,
+            keys.Cast<object>().ToList());
+
+        return this.InternalDeleteBatchAsync(keys, condition, options, cancellationToken);
+    }
+
+    private Task InternalDeleteBatchAsync<TKey>(
+        IEnumerable<TKey> keys,
+        SqliteWhereCondition condition,
+        DeleteRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        const string OperationName = "Delete";
+
+        var tasks = new List<Task>();
+        var keysList = keys.Select(key => (object)key!).ToList();
+
+        if (this._vectorPropertiesExist)
+        {
+            using var vectorCommand = this._commandBuilder.BuildDeleteCommand(
+                this._vectorTableName,
+                [condition]);
+
+            tasks.Add(this.RunOperationAsync(OperationName, () => vectorCommand.ExecuteNonQueryAsync(cancellationToken)));
+        }
+
+        using var dataCommand = this._commandBuilder.BuildDeleteCommand(
+            this._dataTableName,
+            [condition]);
+
+        tasks.Add(this.RunOperationAsync(OperationName, () => dataCommand.ExecuteNonQueryAsync(cancellationToken)));
+
+        return Task.WhenAll(tasks);
```

Comment:
No, because in vector search extension that is used in this implementation, there is no relationship between data table and vector table, which allows us to perform operations against these tables in parallel.